File: pascube_src/pasvulkan/externals/poca/examples/benchmarks/callbench.py
import time

counter = 0

def TestFunction():
  global counter
  counter += 1
  return counter

def TestFunction2():
  global counter
  def TestFunction3():
    global counter
    counter += 1
    return counter
  
  for _ in range(10):
    TestFunction3()
  return counter

# Measure TestFunction performance
start = time.time()
for _ in range(1000000):
  TestFunction()
end = time.time()
elapsed_ms = (end - start) * 1000
print(f"TestFunction Time taken: {elapsed_ms:.2f}ms")

# Measure TestFunction2 performance
start = time.time()
for _ in range(1000000):
  TestFunction2()
end = time.time()
elapsed_ms = (end - start) * 1000
print(f"TestFunction2 Time taken: {elapsed_ms:.2f}ms")

File: pascube_src/pasvulkan/externals/poca/examples/benchmarks/callbench.rb
$counter = 0

def test_function
  $counter += 1
  $counter
end

def test_function2
  test_function3 = lambda do
    $counter += 1
    $counter
  end

  10.times { test_function3.call }
  $counter
end

start_time = Time.now
1_000_000.times { test_function }
end_time = Time.now
elapsed_ms = (end_time - start_time) * 1000.0
puts "TestFunction Time taken: #{elapsed_ms}ms"

start_time = Time.now
1_000_000.times { test_function2 }
end_time = Time.now
elapsed_ms = (end_time - start_time) * 1000.0
puts "TestFunction2 Time taken: #{elapsed_ms}ms"
